lib/trento/application/integration/checks/checks.ex

```diff
@@ -13,9 +12,9 @@ defmodule Trento.Integration.Checks do
 
   require Logger
 
-  @spec request_execution(String.t(), String.t(), Provider.t(), [map], [String.t()]) ::
+  @spec request_execution(String.t(), String.t(), map, [map], [String.t()]) ::
```

Comment:
The only thing, and I'm really thinking out loud here, is if we should maybe treat this as a union type and encode the `env` into a struct that can be `ClusterChecksExecutionEnv | HostChecksExecutionEnv`

Too much? Too verbose? Just shooting the idea

Reply:
Alright, it makes sense, even if its almost only for documentation, as this data is not coming from outside, and we most probably don't need such validation

Reply:
@dottorblaster I did the change. With this approach, the execution is better defined but more rigid at the same time. The end result is the same, and I don't expect the need to have flexible envs (now it is better documented), so this approach is fine for me.
What do you think?